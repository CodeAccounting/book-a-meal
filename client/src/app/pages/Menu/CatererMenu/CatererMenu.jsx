import React, { Component, Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import Modal from '../../../shared/Modal';
import MealCard from '../../../shared/MealCard';
import DatePicker from '../../../shared/DatePicker';
import CatererView from '../../../shared/CatererView';
import { userPropTypes, mealObjPropTypes } from '../../../../helpers/proptypes';

/**
 * @exports
 * @class CatererMenu
 * @extends Component
 * @classdesc Creates CatererMenu Component
 * @returns {JSX} CatererMenu Component
 */
class CatererMenu extends Component {
  static propTypes = {
    ...userPropTypes,
    meals: PropTypes.arrayOf(mealObjPropTypes).isRequired,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchMenu: PropTypes.func.isRequired,
    submitError: PropTypes.string,
    submitting: PropTypes.bool.isRequired,
    toggleModal: PropTypes.func.isRequired,
  }

  static defaultProps = {
    submitError: null
  }

  state = {
    currentDay: moment().format('YYYY-MM-DD')
  }

  /**
 * @memberof CatererMenu
 * @returns {JSX} CatererMenu Component
 */
  componentDidMount() {
    this.fetchMenu();
  }

  /**
   * @memberof CatererMenu
   * @param {string} date
   * @returns {JSX} CatererMenu Component
   */
  updateCurrentDate = (date) => {
    this.setState({
      currentDay: date
    }, () => this.fetchMenu());
  }

  fetchMenu = () => {
    this.props.fetchMenu(this.state.currentDay);
  }

  /**
   * @memberof CatererMenu
   * @returns {JSX} CatererMenu Component
   */
  renderCatererMenu = () => {
    const now = moment().format('YYYY-MM-DD');
    const disableAddButton = moment(now).isAfter(moment(this.state.currentDay)) ? true : null;

    return (
      <Fragment>
        <div className="content-wrapper meals menu-meals">
          <div className="date d-none-md">
            <h2>{moment(this.state.currentDay).format('dddd[,] Do MMMM YYYY')}</h2>
            <DatePicker handleSelectDate={this.updateCurrentDate} />
          </div>
          <DatePicker screenSize="md" handleSelectDate={this.updateCurrentDate} />
          <div className="top">
            <button disabled={disableAddButton} className="btn btn-pri" id="menu-modal-btn" onClick={() => this.props.toggleModal('menu')}>Set Menu</button>
          </div>
          <div className="page-heading">
            <h2>{'Meal Items on This Day\'s Menu'}</h2>
            <hr />
          </div>
          <div className="card-group meals-wrapper" id="card-group">
            {[...this.props.meals].map(meal =>
              (<MealCard
                type="menu"
                key={meal.id}
                meal={meal}
              />))}
          </div>
        </div>
      </Fragment>
    );
  }


  /**
   * @memberof CatererMenu
* @returns {JSX} CatererMenu Component
  */
  render() {
    const {
      user, meals, logout, submitting, submitError, isFetching
    } = this.props;

    return (
      <Fragment>
        <CatererView
          date={moment(this.state.currentDay).format('dddd[,] Do MMMM YYYY')}
          showTime={false}
          user={user}
          logout={logout}
          type="menu"
          isFetching={isFetching}
        >
          {this.renderCatererMenu()}
        </CatererView>
        <Modal
          meals={meals}
          submitting={submitting}
          submitError={submitError}
        />
      </Fragment>
    );
  }
}

export default CatererMenu;
