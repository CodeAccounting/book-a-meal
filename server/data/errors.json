{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}