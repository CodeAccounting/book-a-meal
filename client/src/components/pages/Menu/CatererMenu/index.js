import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import CatererMenu from './CatererMenu';
import { fetchMenu, setCurrentDay } from '../../../../store/operations/menu';
import { logout } from '../../../../store/operations/auth';
import { toggleModal } from '../../../../store/actions/ui';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  meals: state.menu.meals,
  submitting: state.menu.working,
  submitError: state.menu.error,
  currentDay: state.menu.currentDay,
  metadata: state.menu.metadata
});

const mapDispatchToProps = dispatch => bindActionCreators({
  logout, fetchMenu, toggleModal, setCurrentDay
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(CatererMenu);
