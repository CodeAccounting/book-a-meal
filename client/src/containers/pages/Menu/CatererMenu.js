import { connect } from 'react-redux';
import CatererMenu from '../../../components/pages/Menu/CatererMenu';
import { fetchMenu, setCurrentDay } from '../../../actions/menu';
import { logout } from '../../../actions/auth';
import { toggleModal } from '../../../actions/ui';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  meals: state.menu.meals,
  submitting: state.menu.working,
  submitError: state.menu.error,
  currentDay: state.menu.currentDay,
  metadata: state.menu.metadata
});

export default connect(mapStateToProps, {
  logout, fetchMenu, toggleModal, setCurrentDay
})(CatererMenu);
