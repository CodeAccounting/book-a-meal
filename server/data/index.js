import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
