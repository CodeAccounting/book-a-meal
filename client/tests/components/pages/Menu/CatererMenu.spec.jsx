import React from 'react';
import moment from 'moment';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CatererMenu from '../../../../src/components/pages/Menu/CatererMenu/CatererMenu';
import ConnectedCatererMenu from '../../../../src/components/pages/Menu/CatererMenu';
import { caterer, caterersMealsObj, initialValues } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialValues, menu: { ...initialValues.menu, meals: caterersMealsObj.meals }
});
const { now } = Date;

describe('Menu', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const toggleMock = jest.fn();

    const shallowWrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching={false}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('MealCard')).toBeTruthy();
  });

  it('calls toggleModal when set meal button is clicked', () => {
    const toggleMock = jest.fn();
    const wrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching={false}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
    />);

    wrapper.find('#menu-modal-btn').simulate('click');
    expect(toggleMock).toHaveBeenCalled();
  });

  it('calls setCurrentDay when Change Date button is clicked', () => {
    const setCurrentDayMock = jest.fn();
    const wrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching={false}
      submitting={false}
      setCurrentDay={setCurrentDayMock}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={jest.fn()}
    />);

    const event = { target: { value: moment().format('YYYY-MM-DD') } };

    wrapper.find('DatePicker').at(0).dive().find('input')
      .simulate('change', event);
    expect(setCurrentDayMock).toHaveBeenCalled();
  });

  it('renders Preloader when fetching', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders message when not fetching and there are no meals on the menu', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      meals={[]}
      isFetching
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p').text()).toEqual('There are no Meal Items on this Menu');
  });

  it('renders connected component', () => {
    const dispatchMock = jest.fn();
    const comp = (
      <Provider store={store}>
        <ConnectedCatererMenu
          user={caterer}
          dispatch={dispatchMock}
          {...caterersMealsObj}
          isFetching={false}
        />
      </Provider>);

    const wrapper = mount(comp);

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('disbales button when time in state is before current time', () => {
    const toggleMock = jest.fn();
    const wrapper = shallow(<CatererMenu
      user={caterer}
      logout={jest.fn()}
      fetchMenu={jest.fn()}
      {...caterersMealsObj}
      isFetching={false}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
    />);

    wrapper.setProps({
      currentDay: '1969-04-27'
    });

    expect(wrapper.find('#menu-modal-btn').props('disabled')).toBeTruthy();
  });
});
