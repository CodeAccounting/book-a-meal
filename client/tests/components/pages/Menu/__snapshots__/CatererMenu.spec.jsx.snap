// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererMenu renders Preloader when fetching 1`] = `
<React.Fragment>
  <View
    isFetching={true}
    logout={[MockFunction]}
    showTime={false}
    type="menu"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals menu-meals"
      >
        <div
          className="date d-none-md"
        >
          <h2>
            Thursday, 1st January 1970
          </h2>
          <DatePicker
            handleSelectDate={[Function]}
            screenSize={null}
          />
        </div>
        <DatePicker
          handleSelectDate={[Function]}
          screenSize="md"
        />
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            disabled={null}
            id="menu-modal-btn"
            onClick={[Function]}
          >
            Set Menu
          </button>
        </div>
        <div
          className="page-heading"
        >
          <h2>
            Meal Items on This Day's Menu
          </h2>
          <hr />
        </div>
        <div
          className="card-group meals-wrapper"
          id="card-group"
        >
          <MealCard
            key="81211c24-51c0-46ec-b1e0-18db55880958"
            meal={
              Object {
                "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                "price": "1500.00",
                "title": "Jollof Rice, Beef and Plantain",
                "vegetarian": false,
              }
            }
            orderMeal={null}
            toggleModal={null}
            type="menu"
          />
          <MealCard
            key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
            meal={
              Object {
                "description": "Sharwama contains no animal products, perfect for dieters",
                "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                "price": "1200.00",
                "title": "Vegetable Sharwama and Guava Smoothie",
                "vegetarian": true,
              }
            }
            orderMeal={null}
            toggleModal={null}
            type="menu"
          />
        </div>
      </div>
    </React.Fragment>
  </View>
  <Connect(Modal)
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;

exports[`CatererMenu renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CatererMenu)
    isFetching={false}
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <CatererMenu
      currentDay="1970-01-01"
      fetchMenu={[Function]}
      isFetching={false}
      logout={[Function]}
      meals={
        Array [
          Object {
            "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
            "id": "81211c24-51c0-46ec-b1e0-18db55880958",
            "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
            "price": "1500.00",
            "title": "Jollof Rice, Beef and Plantain",
            "vegetarian": false,
          },
          Object {
            "description": "Sharwama contains no animal products, perfect for dieters",
            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
            "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
            "price": "1200.00",
            "title": "Vegetable Sharwama and Guava Smoothie",
            "vegetarian": true,
          },
        ]
      }
      setCurrentDay={[Function]}
      submitError={null}
      submitting={false}
      toggleModal={[Function]}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    >
      <View
        isFetching={false}
        logout={[Function]}
        showTime={false}
        type="menu"
        user={
          Object {
            "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
            "businessName": "Food Circle",
            "businessPhoneNo": "+2348166557788",
            "email": "food@circle.com",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "role": "caterer",
          }
        }
      >
        <div
          className="admin null"
        >
          <Header
            active="menu"
            dateType="menu"
            logout={[Function]}
            showTime={false}
            type="caterer"
            user={
              Object {
                "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                "businessName": "Food Circle",
                "businessPhoneNo": "+2348166557788",
                "email": "food@circle.com",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "role": "caterer",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="navbar"
              >
                <Connect(CatererHeader)
                  dateType="menu"
                  showTime={false}
                >
                  <CatererHeader
                    currentDay="1970-01-01"
                    dateType="menu"
                    showTime={false}
                    toggleSideNav={[Function]}
                  >
                    <div
                      className="menu-control d-none-md"
                    >
                      <Link
                        className="d-none-md"
                        clickHandler={[Function]}
                        id={null}
                      >
                        <button
                          className="link-btn d-none-md"
                          id={null}
                          onClick={[Function]}
                        >
                          <img
                            alt="toggler"
                            src="/img/toggler.png"
                          />
                        </button>
                      </Link>
                    </div>
                    <div
                      className="page-title d-flex-md"
                    >
                      <div
                        className="date"
                      >
                        <h2>
                          Thursday, 1st January 1970
                              
                        </h2>
                      </div>
                    </div>
                  </CatererHeader>
                </Connect(CatererHeader)>
              </div>
            </div>
          </Header>
          <div
            className="content"
          >
            <Connect(SideNav)
              active="menu"
              logout={[Function]}
              user={
                Object {
                  "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "businessPhoneNo": "+2348166557788",
                  "email": "food@circle.com",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "role": "caterer",
                }
              }
            >
              <SideNav
                active="menu"
                logout={[Function]}
                open={false}
                toggleSideNav={[Function]}
                user={
                  Object {
                    "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "businessPhoneNo": "+2348166557788",
                    "email": "food@circle.com",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "role": "caterer",
                  }
                }
              >
                <div
                  className="sidenav null"
                  id="sidenav"
                >
                  <div
                    className="sidenav-content"
                  >
                    <div
                      className="sidenav-header"
                    >
                      <CloseIcon
                        btnID={null}
                        clickHandler={[Function]}
                        divClass="d-none-md"
                      >
                        <div
                          className="close d-none-md"
                        >
                          <Link
                            aria-hidden="true"
                            className={null}
                            clickHandler={[Function]}
                            id={null}
                          >
                            <button
                              className="link-btn null"
                              id={null}
                              onClick={[Function]}
                            >
                              ×
                            </button>
                          </Link>
                        </div>
                      </CloseIcon>
                      <div
                        className="sidenav-title"
                      >
                        <h3>
                          <Link
                            href="/"
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              BOOK-A-MEAL
                            </a>
                          </Link>
                        </h3>
                        <div
                          className="username-circle"
                        >
                          <p>
                            F
                          </p>
                        </div>
                        <p>
                          Food Circle
                        </p>
                        <span>
                          <Link
                            className={null}
                            clickHandler={[Function]}
                            id={null}
                          >
                            <button
                              className="link-btn null"
                              id={null}
                              onClick={[Function]}
                            >
                              Logout
                            </button>
                          </Link>
                        </span>
                      </div>
                    </div>
                    <div
                      className="sidenav-body"
                    >
                      <Link
                        className="menu-item false"
                        href="/"
                        replace={false}
                        to="/"
                      >
                        <a
                          className="menu-item false"
                          href="/"
                          onClick={[Function]}
                        >
                          Dashboard
                        </a>
                      </Link>
                      <Link
                        className="menu-item false"
                        href="/meals"
                        replace={false}
                        to="/meals"
                      >
                        <a
                          className="menu-item false"
                          href="/meals"
                          onClick={[Function]}
                        >
                          Meals
                        </a>
                      </Link>
                      <Link
                        className="menu-item active"
                        href="/menu"
                        replace={false}
                        to="/menu"
                      >
                        <a
                          className="menu-item active"
                          href="/menu"
                          onClick={[Function]}
                        >
                          Menu
                        </a>
                      </Link>
                      <Link
                        className="menu-item false"
                        href="/orders"
                        replace={false}
                        to="/orders"
                      >
                        <a
                          className="menu-item false"
                          href="/orders"
                          onClick={[Function]}
                        >
                          Orders
                        </a>
                      </Link>
                    </div>
                  </div>
                </div>
              </SideNav>
            </Connect(SideNav)>
            <div
              className="content-wrapper"
              id="has-modal"
            >
              <div
                className="content-wrapper meals menu-meals"
              >
                <div
                  className="date d-none-md"
                >
                  <h2>
                    Thursday, 1st January 1970
                  </h2>
                  <DatePicker
                    handleSelectDate={[Function]}
                    screenSize={null}
                  >
                    <div
                      className="datepicker-div"
                    >
                      <button
                        className="btn btn-sec"
                      >
                        Change Date
                      </button>
                      <input
                        id="datepicker"
                        onChange={[Function]}
                        type="date"
                      />
                    </div>
                  </DatePicker>
                </div>
                <DatePicker
                  handleSelectDate={[Function]}
                  screenSize="md"
                >
                  <div
                    className="datepicker-div d-flex-md"
                  >
                    <button
                      className="btn btn-sec"
                    >
                      Change Date
                    </button>
                    <input
                      id="datepicker"
                      onChange={[Function]}
                      type="date"
                    />
                  </div>
                </DatePicker>
                <div
                  className="top"
                >
                  <button
                    className="btn btn-pri"
                    disabled={null}
                    id="menu-modal-btn"
                    onClick={[Function]}
                  >
                    Set Menu
                  </button>
                </div>
                <div
                  className="page-heading"
                >
                  <h2>
                    Meal Items on This Day's Menu
                  </h2>
                  <hr />
                </div>
                <div
                  className="card-group meals-wrapper"
                  id="card-group"
                >
                  <MealCard
                    key="81211c24-51c0-46ec-b1e0-18db55880958"
                    meal={
                      Object {
                        "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                        "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                        "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                        "price": "1500.00",
                        "title": "Jollof Rice, Beef and Plantain",
                        "vegetarian": false,
                      }
                    }
                    orderMeal={null}
                    toggleModal={null}
                    type="menu"
                  >
                    <div
                      className="meal-card false"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
                        />
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Jollof Rice, Beef and Plantain
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1500.00
                          </h3>
                          <p>
                            Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
                          </p>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                  <MealCard
                    key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                    meal={
                      Object {
                        "description": "Sharwama contains no animal products, perfect for dieters",
                        "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                        "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                        "price": "1200.00",
                        "title": "Vegetable Sharwama and Guava Smoothie",
                        "vegetarian": true,
                      }
                    }
                    orderMeal={null}
                    toggleModal={null}
                    type="menu"
                  >
                    <div
                      className="meal-card false"
                      id="meal-card"
                    >
                      <div
                        className="meal-card-header"
                      >
                        <img
                          alt="meal"
                          src="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
                        />
                        <div
                          className="menu-card-title"
                        >
                          <p>
                            Vegetable Sharwama and Guava Smoothie
                          </p>
                        </div>
                      </div>
                      <div
                        className="meal-card-body"
                      >
                        <div>
                          <h3>
                            ₦ 
                            1200.00
                          </h3>
                          <p>
                            Sharwama contains no animal products, perfect for dieters
                          </p>
                        </div>
                      </div>
                    </div>
                  </MealCard>
                </div>
              </div>
            </div>
          </div>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </View>
      <Connect(Modal)
        meals={
          Array [
            Object {
              "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
              "id": "81211c24-51c0-46ec-b1e0-18db55880958",
              "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
              "price": "1500.00",
              "title": "Jollof Rice, Beef and Plantain",
              "vegetarian": false,
            },
            Object {
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ]
        }
        submitError={null}
        submitting={false}
      >
        <Modal
          meals={
            Array [
              Object {
                "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                "price": "1500.00",
                "title": "Jollof Rice, Beef and Plantain",
                "vegetarian": false,
              },
              Object {
                "description": "Sharwama contains no animal products, perfect for dieters",
                "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                "price": "1200.00",
                "title": "Vegetable Sharwama and Guava Smoothie",
                "vegetarian": true,
              },
            ]
          }
          open={false}
          submitError={null}
          submitting={false}
          toggleModal={[Function]}
          type={null}
        />
      </Connect(Modal)>
    </CatererMenu>
  </Connect(CatererMenu)>
</Provider>
`;

exports[`CatererMenu renders correctly when not fetching 1`] = `
<React.Fragment>
  <View
    isFetching={false}
    logout={[MockFunction]}
    showTime={false}
    type="menu"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals menu-meals"
      >
        <div
          className="date d-none-md"
        >
          <h2>
            Thursday, 1st January 1970
          </h2>
          <DatePicker
            handleSelectDate={[Function]}
            screenSize={null}
          />
        </div>
        <DatePicker
          handleSelectDate={[Function]}
          screenSize="md"
        />
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            disabled={null}
            id="menu-modal-btn"
            onClick={[Function]}
          >
            Set Menu
          </button>
        </div>
        <div
          className="page-heading"
        >
          <h2>
            Meal Items on This Day's Menu
          </h2>
          <hr />
        </div>
        <div
          className="card-group meals-wrapper"
          id="card-group"
        >
          <MealCard
            key="81211c24-51c0-46ec-b1e0-18db55880958"
            meal={
              Object {
                "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                "price": "1500.00",
                "title": "Jollof Rice, Beef and Plantain",
                "vegetarian": false,
              }
            }
            orderMeal={null}
            toggleModal={null}
            type="menu"
          />
          <MealCard
            key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
            meal={
              Object {
                "description": "Sharwama contains no animal products, perfect for dieters",
                "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                "price": "1200.00",
                "title": "Vegetable Sharwama and Guava Smoothie",
                "vegetarian": true,
              }
            }
            orderMeal={null}
            toggleModal={null}
            type="menu"
          />
        </div>
      </div>
    </React.Fragment>
  </View>
  <Connect(Modal)
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageURL": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;

exports[`CatererMenu renders message when not fetching and there are no meals on the menu 1`] = `
<React.Fragment>
  <View
    isFetching={true}
    logout={[MockFunction]}
    showTime={false}
    type="menu"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <React.Fragment>
      <div
        className="content-wrapper meals menu-meals"
      >
        <div
          className="date d-none-md"
        >
          <h2>
            Thursday, 1st January 1970
          </h2>
          <DatePicker
            handleSelectDate={[Function]}
            screenSize={null}
          />
        </div>
        <DatePicker
          handleSelectDate={[Function]}
          screenSize="md"
        />
        <div
          className="top"
        >
          <button
            className="btn btn-pri"
            disabled={null}
            id="menu-modal-btn"
            onClick={[Function]}
          >
            Set Menu
          </button>
        </div>
        <div
          className="page-heading"
        >
          <h2>
            Meal Items on This Day's Menu
          </h2>
          <hr />
        </div>
        <p
          className="text-center"
        >
          There are no Meal Items on this Menu
        </p>
      </div>
    </React.Fragment>
  </View>
  <Connect(Modal)
    meals={Array []}
    submitError={null}
    submitting={false}
  />
</React.Fragment>
`;
